File: apps/api/src/videos/entities/video.entity.ts
import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';

@Entity({ name: 'videos' })
export class Video {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'youtube_id', nullable: false, unique: true })
  youtubeId: string;

  @Column({ type: 'text', nullable: false })
  title: string;

  @Column({ type: 'text', nullable: true })
  description: string;

  @Column({ name: 'thumbnail_url', type: 'text', nullable: true })
  thumbnailUrl: string;

  @Column({ type: 'bigint', nullable: true, comment: 'duration in seconds' })
  duration: number;

  @Column({ name: 'uploader_id', type: 'bigint', nullable: true })
  uploaderId: number;

  @Column({ name: 'channel_title', type: 'text', nullable: true })
  channelTitle: string;

  @Column({ type: 'jsonb', nullable: true })
  metadata: Record<string, any>;

  @Column({
    name: 'created_at',
    type: 'timestamp',
    default: () => 'CURRENT_TIMESTAMP',
  })
  createdAt: Date;
}

File: apps/api_go/internal/domain/video_entity.go
package domain

import (
	"encoding/json"

	"gorm.io/gorm"
)

type Video struct {
	gorm.Model
	YoutubeID    string          `gorm:"column:youtube_id;unique;not null"`
	Title        string          `gorm:"column:title;type:text;not null"`
	Description  *string         `gorm:"column:description;type:text"`
	ThumbnailURL *string         `gorm:"column:thumbnail_url;type:text"`
	Duration     *int64          `gorm:"column:duration;type:bigint"` // duration in seconds
	UploaderID   *uint           `gorm:"column:uploader_id;type:bigint"`
	ChannelTitle *string         `gorm:"column:channel_title;type:text"`
	Metadata     json.RawMessage `gorm:"column:metadata;type:jsonb"`
}

func (Video) TableName() string {
	return "videos"
}
